bin/improver-ensemble-calibration

```diff
@@ -0,0 +1,128 @@
+
+
+def main():
+    """Load in arguments and get going."""
+    parser = argparse.ArgumentParser(
+    description='Apply the requested ensemble calibration method using '
+                'historical forecast and "truth" data.')
+    parser.add_argument('calibration_method',
+                        metavar='ENSEMBLE_CALIBRATION_METHOD',
+                        choices=['ensemble model output statistics',
+                                 'nonhomogeneous gaussian regression'],
+                        help='The calibration method that will be applied. '
+                             'Supported methods are: "emos" (ensemble model '
+                             'output statistics) and "ngr" (nonhomogeneous '
+                             'gaussian regression).')
+    parser.add_argument('units', metavar='UNITS_TO_CALIBRATE_IN',
+                        help='The unit that calibration should be undertaken '
+                             'in. The current forecast, historical forecast '
+                             'and truth will be converted as required.')
+    parser.add_argument('distribution', metavar='DISTRIBUTION',
+                        choices=['gaussian', 'truncated gaussian'],
+                        help='The distribution that will be used for '
+                             'calibration. This will be dependent upon the '
+                             'input phenomenon. This has to be supported by '
+                             'the minimisation functions in '
+                             'ContinuousRankedProbabilityScoreMinimisers.')
+
+    parser.add_argument('input_filepath', metavar='INPUT_FILE',
+                        help='A path to an input NetCDF file containing the '
+                             'current forecast to be processed.')
+    parser.add_argument('historic_filepath', metavar='HISTORIC_DATA_FILE',
+                        help='A path to an input NetCDF file containing the '
+                             'historic forecast used for calibration.')
+    parser.add_argument('truth_filepath', metavar='TRUTH_DATA_FILE',
+                        help='A path to an input NetCDF file containing the '
+                             'historic truth analyses used for calibration.')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
+                        help='The output path for the processed NetCDF')
+
+    parser.add_argument('--predictor_of_mean', metavar='CALIBRATE_MEAN_FLAG',
+                        choices=['mean', 'members'], default='mean',
+                        help='String to specify the input to calculate the '
+                             'calibrated mean. Currently the ensemble mean '
+                             '("mean") and the ensemble members ("members") '
+                             'are supported as the predictors. Default: '
+                             '"mean".')
+    parser.add_argument('--save_mean_variance', metavar='MEAN_VARIANCE_FILE',
```

Comment:
I think it might be necessary to add an option to switch on and off the Ensemble Copula Coupling i.e. the `GeneratePercentilesFromMeanAndVariance` and `EnsembleReordering` plugins. I think usually you'd want to run ensemble calibration and ECC, however, it might be worth supporting this option. This links with the `--save_mean_variance` option, because I don't think you'd want to run ensemble calibration without ECC and not save the output from the ensemble calibration.

The typical route is:
- Run ensemble calibration, run ECC, save ECC output.

Currently, you're also supporting:
- Run ensemble calibration, save output, run ECC, save ECC output
 
I've thought about these options:
- Run ensemble calibration - this option is a bit pointless.
- Run ensemble calibration, save output - I think only this needs to be supported.